src/data/nat/factorization.lean

```diff
@@ -196,4 +199,62 @@ begin
   { rintro ⟨c, rfl⟩, rw factorization_mul hd (right_ne_zero_of_mul hn), simp },
 end
 
+lemma prime_pow_dvd_iff_factorization_ge (p k n : ℕ) (pp : prime p) (hn : n ≠ 0) :
+  p ^ k ∣ n ↔ k ≤ n.factorization p :=
+by rw [←factorization_le_iff_dvd (pow_pos pp.pos k).ne' hn, pp.factorization_pow, single_le_iff]
+
+lemma exists_factorization_lt_of_lt {a b : ℕ} (ha : a ≠ 0) (hb : b ≠ 0) (hab : a < b) :
+  ∃ p : ℕ, a.factorization p < b.factorization p :=
+begin
+  contrapose! hab,
+  rw [←finsupp.le_def, factorization_le_iff_dvd hb ha] at hab,
+  exact le_of_dvd ha.bot_lt hab,
+end
+
+@[simp]
+lemma div_factorization_eq_tsub_of_dvd {d n : ℕ} (hd : d ≠ 0) (hn : n ≠ 0) (h : d ∣ n) :
+  (n/d).factorization = n.factorization - d.factorization :=
+begin
+  cases dvd_iff_exists_eq_mul_left.mp h with c hc,
+  have hc_pos : c ≠ 0, { subst hc, exact left_ne_zero_of_mul hn },
+  rw [hc, nat.mul_div_cancel c hd.bot_lt, factorization_mul hc_pos hd, add_tsub_cancel_right],
+end
+
+lemma dvd_iff_div_factorization_eq_tsub (d n : ℕ) (hd : d ≠ 0) (hn : n ≠ 0) (hdn : d ≤ n) :
```

Comment:
Similarly to before, `hdn` and `hd` should imply `hn`

Reply:
Does that work in this case, since here we have `≤` rather than `<`?

Reply:
But I guess this should hold for `n=0` as well, so I can deal with that with an `rcases` line.

Reply:
Yeah it still should - we have 0 < d and d <= n so we can deduce 0 < n.

Reply:
Oh of course, how stupid! This I what I get for commenting as I'm about to fall asleep!